hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,391 @@ This document aims to make the distinction between physical entropy and factory
 However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
 That is either physical entropy, or the output of a DRNG which itself has been seeded (and possibly reseeded) with physical entropy.
 In CC terms, "entropy strings" (when used in this document without a qualifier) should be understood to come from either a PTG.2 or PTG.3 class RNG.
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+The CSRNG block has been constructed to follow the NIST recommendation for a DRBG mechanism based on block ciphers.
+Specifically, it is a CTR_DRBG that uses an approved block cipher algorithm in counter mode.
+As such, the block diagram below makes reference to hardware blocks that either directly or closely follow NIST descriptions for the equivalent functions.
+
+There are two major hardware interfaces: the application interface and the entropy request interface.
+The application interface, which is described in more detail later, is provided for an application to manage an `instance` in CSRNG.
+Once setup, the application interface user can request for entropy bits to be generated, as well as other functions.
+The application interface supports up to 15 hardware interfaces, and one software interface.
+
+A walk through of how CSNRG generates entropy bits begins with the application interface.
```

Comment:
this description is awesome.

Reply:
and I meant to say earlier, thanks for this review @tjaychen !